scene/2d/physics_body_2d.cpp

```diff
@@ -1386,6 +1398,18 @@ void CharacterBody2D::_move_and_slide_floating(double p_delta) {
 
 		first_slide = false;
 	}
+
+	// Scales the velocity according to the wall slope.
```

Comment:
Why do you want to do this? In floating mode, in 2D, we are in top down situation where there is no concept of "slope".

Reply:
Well, by "slope" I just meant whatever wall we hit.  However, it seems it's no longer necessary with how I rewrote the earlier parts, so I've removed it.